core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -577,6 +578,13 @@ public int getWaitingSplits()
         return waitingSplits.size();
     }
 
+    @Managed
+    @Nested
+    public DistributionStat getWaitingSplitsSizeDistribution()
```

Comment:
`waitingSplits` are splits that are waiting for executor thread.
You should rather monitor size of `io.trino.execution.executor.TaskExecutor#allSplits`, so that we can see if worker has enough splits or not.